### ISSUE TITLE
Error when making call from client to server

### ISSUE THREAD

**Issue by tn222eb** [GetStream/stream-php#15]
Fatal error: Uncaught exception 'GuzzleHttp\Ring\Exception\RingException' with message 'cURL error 60: SSL certificate problem: self signed certificate in certificate chain' in C:\workspace\app\vendor\guzzlehttp\ringphp\src\Client\CurlFactory.php:126 Stack trace: #0 C:\workspace\app\vendor\guzzlehttp\ringphp\src\Client\CurlFactory.php(90): GuzzleHttp\Ring\Client\CurlFactory::createErrorResponse(Object(GuzzleHttp\Ring\Client\CurlHandler), Array, Array) #1 C:\workspace\app\vendor\guzzlehttp\ringphp\src\Client\CurlHandler.php(78): GuzzleHttp\Ring\Client\CurlFactory::createResponse(Object(GuzzleHttp\Ring\Client\CurlHandler), Array, Array, Array, Resource id #68) #2 C:\workspace\app\vendor\guzzlehttp\ringphp\src\Client\Middleware.php(54): GuzzleHttp\Ring\Client\CurlHandler->__invoke(Array) #3 C:\workspace\app\vendor\guzzlehttp\ringphp\src\Client\Middleware.php(30): GuzzleHttp\Ring\Client\Middleware::GuzzleHttp\Ring\Client{closure}(Array) #4 C:\workspace\app\vendor\guzzlehttp\guzzle\src\RequestFsm.php(129): GuzzleHttp\Ring\Cli in C:\workspace\app\vendor\guzzlehttp\guzzle\src\Exception\RequestException.php on line 51

I would appreciate if you could help me get around this error

**Comment by tbarbugli** [comment]
Hi,

Thank you for reaching us. It seems like the certificate keychain used by
php curl is out of date or missing (and/or you do not have openSSL
installed). I suggest you to follow the steps from this guide (section 3 is
the most relevant) http://support.gnip.com/articles/curl-on-win7.html

Please let us know if you need more help with this issue.

Best,
Tommaso

**Comment by tn222eb** [comment]
I have followed your steps and also searched google for further help but to
no success.

Can you please help me, the api still dont work

2015-04-01 14:53 GMT+02:00 Tommaso Barbugli <notifications@github.com>:

**Comment by davgothic** [comment]
1, Download the [Certficate Authority Bundle](http://curl.haxx.se/ca/cacert.pem) and save it somewhere locally.
2, Instantiate a feed object like so `$user_feed = $client->feed('user', '1');`
3, Add a default option to the Guzzle HTTP client for that feed like so `$user_feed->setGuzzleDefaultOption('verify', '/absolute/path/to/cacert.pem');` This will have to be done for every feed you instantiate because as far as I am aware GetStream haven't provided us with a method to set a global option on the HTTP client or a method of passing our own in.

Full example:

    // Instantiate a feed object
    $user_feed = $client->feed('user', '1');
    
    // You have to manually set the path to the cacert.pem. This must be absolute.
    $user_feed->setGuzzleDefaultOption('verify', '/absolute/path/to/cacert.pem');

    $feed->addActivity([
        'actor' => '\Inviite\Model\User:2',
        'verb' => 'review',
        'object' => 'Review:'.$review->id,
    ]);

I hope this helps somebody.

**Comment by tbarbugli** [comment]
David's suggestion is correct. I am going to add support for options to be
defined at client level and add mention this specific use use case in the
readme.

**Comment by MasterB** [comment]
hi, no @davgothic it's not solve my problem. thanks.
on guzzle 5, there was no problem with it.

**Comment by tbarbugli** [comment]
@MasterB can you elaborate a bit more?

**Comment by MasterB** [comment]
@tbarbugli I've updated Guzzle from 5 to 6. Now its not possible to connect to my https - API, which has a Sneakoil certificate installed. I use PHP 5.5.x. Maybe there is a config problem with the server.

I've tried some things without success. Simple 'verify' => false no longer works for me.

http://docs.guzzlephp.org/en/latest/request-options.html#verify-option

I'm not an SSL certifacte expert.

**Comment by sabrinapisol** [comment]
I've seen on the class GetStream\Stream\Feed that a new GuzzleHttp\Psr7\Request is made, and it has $this->guzzleOptions as fifth parameter. 
Watching to the class GuzzleHttp\Psr7\Request I see that the constructor don't expects to receive that parameter there. And I think that because of that the method setGuzzleDefaultOption do nothing.

**Comment by hannesvdvreken** [comment]
@sabrinapisol You're right! I created this PR #45 which should fix that! It's probably a remainder of upgrading from Guzzle v4-5 to Guzzle v6.

**Comment by hannesvdvreken** [comment]
@MasterB also for you, #45 should fix that. Could you let me know if it works if you install it using

```
composer require get-stream/stream:"dev-fix/guzzle-request-options"
```
(please don't commit that, but only run it locally)

and then execute the usual

```php
$feed->setGuzzleDefaultOption('verify', '/absolute/path/to/cacert.pem');
// or:
$feed->setGuzzleDefaultOption(GuzzleHttp\RequestOptions::VERIFY, '/absolute/path/to/cacert.pem');
// see: http://docs.guzzlephp.org/en/stable/request-options.html#verify
```

Thanks!

**Comment by hannesvdvreken** [comment]
@sabrinapisol @MasterB Replaced #45 with #49.

**Comment by hannesvdvreken** [comment]
Fixed in 2.4.1
